### PR TITLE
adding javascript references to dotnet views based on webpack config …

### PR DESCRIPTION
**Pull request by artmasa**
webpack configuration has changed and it has broken the dotnet template for aurelia.
you can find details on issue #1014

### CODE REVIEW

File: lib/resources/content/javascriptservices/Views/Shared/_Layout.cshtml [hunk 1]
```diff
@@ -4,8 +4,7 @@
         <meta charset="utf-8">
         <meta name="viewport" content="width=device-width, initial-scale=1.0">
         <title>@ViewData["Title"] - Aurelia</title>
-
-        <link rel="stylesheet" href="~/dist/vendor.css" asp-append-version="true">
```
Comment (artmasa) [inline lib/resources/content/javascriptservices/Views/Shared/_Layout.cshtml]: index.ejs states the styles will be added automatically here: https://github.com/artmasa/cli/blob/master/lib/resources/content/index-webpack.ejs

**Comment by huochunpeng** [comment]
cc @brandonseydel

**Comment by artmasa** [comment]
> cc @brandonseydel

can a team member review these changes?

**Comment by jmoerdyk** [comment]
Any word on integrating this?  This would fix the issue I reported in #1014

**Comment by huochunpeng** [comment]
@brandonseydel pls.

File: lib/resources/content/javascriptservices/Views/Shared/_Layout.cshtml (hunk 1 above)
Comment (brandonseydel) [inline lib/resources/content/javascriptservices/Views/Shared/_Layout.cshtml]: Can you glob this so it won't fail if no vendor css I don't like seeing the 404 out the gate.  After that it all looks good.
<link rel="stylesheet" asp-append-version="true" asp-href-include="~/dist/vendor*.css">

**Comment by huochunpeng** [comment]
On current master, the affected files are moved to 
```
skeleton/dotnet-core/Views/Home/Index.cshtml__skip-if-exists
skeleton/dotnet-core/Views/Home/Index.cshtml__instructions
skeleton/dotnet-core/Views/Shared/_Layout.cshtml__skip-if-exists
skeleton/dotnet-core/Views/Shared/_Layout.cshtml__instructions
```

You would need to update the instruction files as well, when user has existing Index.cshtml or _Layout.cshtml file, aurelia-cli will not override them, instead cli will print out instructions for user to update that two files manually.

The instructions are also saved into file `instructions.txt` in the target project folder.

**Comment by huochunpeng** [comment]
@artmasa would you like to update this PR?

File: lib/resources/content/javascriptservices/Views/Shared/_Layout.cshtml (hunk 1 above)
Comment (arjendeblok) [reply lib/resources/content/javascriptservices/Views/Shared/_Layout.cshtml]: Or if you also use app*.css files.

`<link rel="stylesheet" asp-append-version="true" asp-href-include="~/dist/css/*.css"/>`

**Comment by artmasa** [comment]
@huochunpeng I will put some time this weekend to get those updates in.

**Comment by 3cp** [comment]
@artmasa just a reminder. We hope to have this fix for v1 release.

**Comment by artmasa** [comment]
@brandonseydel and @3cp, I have made the modifications to files in the new skeleton folder.
@3cp, you are referring to `instructions.txt` file in your comment. I was not able to find such file in the source. Is this an auto-gened file?

**Comment by 3cp** [comment]
Yes, `instructions.txt` is the generated files with all applied `__instructions` files.
FYI, you can test your branch with `npx artmasa/cli#donet-js-refs new`.

@brandonseydel would you help to do a final review?

**Comment by arjendeblok** [comment]
The line 
`<script type="text/javascript" asp-src-include="~/dist/app*.chunk.js"></script>`
does not work correctly if you are using async chunks. 
Async chunks are generated when you specify e.g. different modules in your app router, so you do not have to load everything at startup.
In this line of code the asynchronous chunks are also loaded at startup. 
For the initial creation of a web application is may be o.k. but it may be in the documentation or instructions somehow.

**Comment by artmasa** [comment]
@arjendeblok, so adding a comment on this line to instruct the developer to remove this line when using async chunks will suffice?

**Comment by arjendeblok** [comment]
@artmasa in my project i use ` <script type="text/javascript" asp-src-include="~/dist/app~*.chunk.js"></script>` So with a `~` character after `app`. And in my router i use e.g. `app.feature1`. This will generate a script with `app.feature1.~` so it will not be included.

I use the same solution for vendor and common.

**Comment by 3cp** [comment]
I don't know about dotnet and the `~` thing :-) @artmasa let us know when it's ready to merge, thanks!

**Comment by artmasa** [comment]
@arjendeblok, `~` worked for me only with the `app~*.chunk.js` applying it to `vendor` or `vendors` did not work. Application never started for me. I think at this point is ready to merge unless a comment is necessary above this line to recommend `~` to developers when wanting async chunks. If this comment is not necessary we are ready to merge @3cp.

**Comment by 3cp** [comment]
@EisenbergEffect can you have a read on the async thing above? We can merge now and improve later, or wait for some improvement.

**Comment by arjendeblok** [comment]
I am using 
```
<script type="text/javascript" asp-src-include="~/dist/runtime*.bundle.js"></script>
<script type="text/javascript" asp-src-include="~/dist/vendor~*.chunk.js"></script>
<script type="text/javascript" asp-src-include="~/dist/vendors~*.chunk.js"></script>
<script type="text/javascript" asp-src-include="~/dist/app~*.chunk.js"></script>
```
And that is working fine. The Common part is always loaded asynchronously, so it does not have to be included.

**Comment by arjendeblok** [comment]
See https://github.com/arjendeblok/aurelia-webpack-chunk-test for an example.

**Comment by arjendeblok** [comment]
Found a better solution by excluding explicitly the async chunks.
```
<script type="text/javascript" asp-src-include="~/dist/runtime*.bundle.js"></script>
<script type="text/javascript" asp-src-include="~/dist/vendor*.chunk.js"
                               asp-src-exclude="~/dist/vendor*.async*.chunk.js"></script>
<script type="text/javascript" asp-src-include="~/dist/app*.chunk.js"
                               asp-src-exclude="~/dist/app*.async*.chunk.js"></script>
```

**Comment by arjendeblok** [comment]
@EisenbergEffect any update on this.
It is not a good impression when you start the CLI, pick ASP.NET Core and the generated project will not start.

**Comment by artmasa** [comment]
@arjendeblok I have tested your last suggestion and have included it in this pull request

**Comment by EisenbergEffect** [comment]
All merged in. Should go out in a release this week.

**Pull request merged**